get_global_param.c

```diff
@@ -102,11 +102,41 @@ void ProgramState::initGrid(const std::vector<cell_info_struct>& cells) {
         }
       }
     }
-    global_param.gridNumLatDivisions = ((largestLat - global_param.gridStartLat) / global_param.gridStepLat) + 1;
-    global_param.gridNumLonDivisions = ((largestLon - global_param.gridStartLon) / global_param.gridStepLon) + 1;
+    global_param.gridNumLatDivisions = ((global_param.gridEndLat - global_param.gridStartLat) / global_param.gridStepLat) + 1;
+    global_param.gridNumLonDivisions = ((global_param.gridEndLon - global_param.gridStartLon) / global_param.gridStepLon) + 1;
   }
 }
 
+void ProgramState::initCellMask(const std::vector<cell_info_struct>& cells) {
+
+  char            ErrStr[MAXSTRING];
+	int total_num_cells = global_param.gridNumLatDivisions * global_param.gridNumLonDivisions;
+	valid_cell_mask = new bool [total_num_cells];
```

Comment:
Is there a good reason for the mask to be a simple array rather than being two dimensional? Ultimately, it's the same information, but it just is weird to have the mask disagree in structure with everything else about the grid. If it were 2d, then the iteration matches up with both of your loops, whereas as you've written it, it's awkward to only increment your index inside the inner loop.